nsq/async.py

```diff
@@ -468,6 +468,9 @@ def _on_data(self, data, **kwargs):
             self.in_flight += 1
 
             message = protocol.decode_message(data)
+            if self.msg_timeout:
+                message._set_timeout(self)
```

Comment:
As an alternative here we could create a registry for the timeouts here... then we wouldn't have to incorporate them into the state of the message. Another not-so-nice alternative would be to make the method public.